src/sinks/pulsar/config.rs

```diff
@@ -0,0 +1,292 @@
+        Healthcheck, VectorSink,
+    },
+};
+use codecs::{encoding::SerializerConfig, TextSerializerConfig};
+use futures_util::FutureExt;
+use pulsar::authentication::oauth2::{OAuth2Authentication, OAuth2Params};
+use pulsar::error::AuthenticationError;
+use pulsar::{
+    compression, message::proto, Authentication, Error as PulsarError, ProducerOptions, Pulsar,
+    TokioExecutor,
+};
+use snafu::ResultExt;
+use value::Kind;
+use vector_common::sensitive_string::SensitiveString;
+use vector_config::configurable_component;
+use vector_core::config::DataType;
+
+/// Configuration for the `pulsar` sink.
+#[configurable_component(sink("pulsar"))]
+#[derive(Clone, Debug)]
+pub struct PulsarSinkConfig {
+    /// The endpoint to which the Pulsar client should connect to.
+    ///
+    /// The endpoint should specify the pulsar protocol and port.
+    #[serde(alias = "address")]
+    #[configurable(metadata(docs::examples = "pulsar://127.0.0.1:6650"))]
+    pub(crate) endpoint: String,
+
+    /// The Pulsar topic name to write events to.
+    #[configurable(metadata(docs::examples = "topic-1234"))]
+    pub(crate) topic: String,
+
+    /// The name of the producer. If not specified, Pulsar will generate a unique name.
+    #[configurable(metadata(docs::examples = "producer-name"))]
+    pub(crate) producer_name: Option<String>,
+
+    /// The log field name or tags key to use for the topic key.
+    ///
+    /// If the field does not exist in the log or in tags, a blank value will be used. If unspecified, the key is not sent.
+    ///
+    /// Pulsar uses a hash of the key to choose the topic-partition or uses round-robin if the record has no key.
+    #[configurable(metadata(docs::examples = "message"))]
+    #[configurable(metadata(docs::examples = "my_field"))]
+    pub(crate) partition_key_field: Option<String>,
+
+    /// The log field name to use for the Pulsar properties.
+    ///
+    /// If omitted, no properties will be written.
+    pub properties_key: Option<String>,
+
+    #[configurable(derived)]
+    #[serde(default)]
+    pub(crate) batch: PulsarBatchConfig,
+
+    #[configurable(derived)]
+    #[serde(default)]
+    pub compression: PulsarCompression,
+
+    #[configurable(derived)]
+    pub encoding: EncodingConfig,
+
+    #[configurable(derived)]
+    pub(crate) auth: Option<PulsarAuthConfig>,
+
+    #[configurable(derived)]
+    #[serde(
+        default,
+        deserialize_with = "crate::serde::bool_or_struct",
+        skip_serializing_if = "crate::serde::skip_serializing_if_default"
+    )]
+    pub acknowledgements: AcknowledgementsConfig,
+}
+
+/// Event batching behavior.
+#[configurable_component]
+#[derive(Clone, Copy, Debug, Default)]
+pub(crate) struct PulsarBatchConfig {
+    /// The maximum size of a batch before it is flushed.
+    #[configurable(metadata(docs::type_unit = "events"))]
+    #[configurable(metadata(docs::examples = 1000))]
+    pub max_events: Option<u32>,
```

Comment:
Should we note that this has a lower limit than usual, as I think we generally use `u64`?